vtex-apps/slider-layout
# Add new 'centerMode' feature

#### What problem is this solving?

We don't currently support sliders that behave like this:

<img src="https://user-images.githubusercontent.com/27777263/90565876-ef4a3180-e17d-11ea-8870-c3a030ca471d.png" width="400px" />

This is something users have been asking for a long time, this PR enables us to do so, through the use of a new `centerMode` prop.

Also, we now support this version as well (setting `centerMode` to `to-the-left`):

<img src="https://user-images.githubusercontent.com/27777263/90566196-74354b00-e17e-11ea-9025-ffb23c506dec.png" width="400px" />

#### How to test it?

Check this [Workspace](https://sliderlayoutcentermode--storecomponents.myvtex.com/) out. You should see that the shelf looks normal on a desktop. Switch to mobile and you can see the new behavior, where there is a "preview" of the next and previous items.

Just to illustrate the case where I think this feature works best, I used the following `slider-layout` props to achieve the behavior shown in the workspace.

```json
{
  "slider-layout#demo-products": {
    "props": {
      "itemsPerPage": {
        "desktop": 5,
        "tablet": 3,
        "phone": 1
      },
      "infinite": true,
      "fullWidth": false,
      "blockClass": "shelf",
      "showNavigationArrows": "desktopOnly",
      "centerMode": {
        "desktop": "disabled",
        "mobile": "center"
      }
    }
  },
}
```

#### Screenshots or example usage:

**Phone**
<img width="358" alt="Screen Shot 2020-08-18 at 18 24 28" src="https://user-images.githubusercontent.com/27777263/90567140-16096780-e180-11ea-9bf0-f884271d0eda.png">

**Tablet**
<img width="571" alt="Screen Shot 2020-08-18 at 18 24 01" src="https://user-images.githubusercontent.com/27777263/90567144-19045800-e180-11ea-91dd-c5cac1b27d87.png">

**Desktop**
<img width="1554" alt="Screen Shot 2020-08-18 at 18 22 41" src="https://user-images.githubusercontent.com/27777263/90567148-1a358500-e180-11ea-8fec-5b629e9c5351.png">

#### How does this PR make you feel? [:link:](http://giphy.com/)



![Math](https://media.giphy.com/media/BmmfETghGOPrW/giphy.gif)
